src/handlers.ts

```diff
@@ -304,6 +304,101 @@ export async function replyAction(
     });
 }
 
+export function createUnfollowActionHandler(accountService: AccountService) {
+    return async function unfollowAction(
+        ctx: Context<{ Variables: HonoContextVariables }>,
+    ) {
+        const handle = ctx.req.param('handle');
+        const apCtx = fedify.createContext(ctx.req.raw as Request, {
+            db: ctx.get('db'),
+            globaldb: ctx.get('globaldb'),
+            logger: ctx.get('logger'),
+        });
+
+        const actorToUnfollow = await lookupObject(apCtx, handle);
+
+        if (!isActor(actorToUnfollow)) {
+            return new Response(null, {
+                status: 404,
+            });
+        }
+
+        const account = await accountService.getDefaultAccountForSite(
+            ctx.get('site'),
+        );
+
+        if (actorToUnfollow.id!.href === account.ap_id) {
+            return new Response(null, {
+                status: 400,
+            });
+        }
```

Comment:
Do we need to account for this? We already have logic that prevents us from following ourself right?